hypershift-operator/controllers/hci/aws_infra.go

```diff
@@ -0,0 +1,169 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hostedclusterinfrastructure
+
+import (
+	awssdk "github.com/aws/aws-sdk-go/aws"
+	"github.com/openshift/hypershift/api/util/ipnet"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/cmd/infra/aws"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Returns true when a requeue is required and pass an error if it is available
```

Comment:
I'm struggling to follow this description, I'd expect something named `createAws` to do one thing and create something in AWS? seems like this needs to be broken down

Reply:
So createAwsIam was originally called from the reconcile loop, but because only AWS had IAM, I decided to put it in generic createAws.

So it could be renamed createAwsInfraIam if you feel strongly about it? @albarbaro

Reply:
no longer applies to this pr.